### PR TITLE
build(feat): More make targets reporting to Sentry.io

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -208,20 +161,21 @@ lint-js:
         create-db \
         apply-migrations \
         reset-db \
-        setup-pyenv \
-        setup-git-config \
```
Comment (armenzg) [inline Makefile:212]: I've reordered these so I can visually compare the make targets in the file and making sure that we're not missing any of them.

File: Makefile [hunk 2]
```diff
@@ -231,6 +185,7 @@ lint-js:
         test-python \
         test-python-ci \
         test-snuba \
+        backend-typing \
```
Comment (armenzg) [inline Makefile:188]: This one was missing.

File: scripts/lib.sh [hunk 3]
```diff
@@ -7,6 +7,15 @@ require() {
     command -v "$1" >/dev/null 2>&1
 }
 
+configure-sentry-cli() {
```
Comment (armenzg) [inline scripts/lib.sh:10]: This way `pyenv.sh` and `do.sh` can both use this code.

File: scripts/lib.sh [hunk 4]
```diff
@@ -39,19 +48,6 @@ install-py-dev() {
     fi
 }
 
-init-config() {
```
Comment (armenzg) [inline scripts/lib.sh:42]: Moving these three closer to `bootstrap` since they're used by that function.

File: Makefile [hunk 5]
```diff
@@ -1,73 +1,29 @@
...
+create-db \
+apply-migrations \
+reset-db \
+setup-git \
+node-version-check \
+install-js-dev \
+install-py-dev :
+	@./scripts/do.sh $@
+
+build-platform-assets \
```
Comment (armenzg) [inline Makefile:19]: These 3 targets are executed in the CI or direnv when there's not a loaded venv, thus, using `SENTRY_NO_VENV_CHECK=1`

File: Makefile [hunk 6]
```diff
@@ -1,73 +1,29 @@
...
-
-create-db:
-	@echo "--> Creating 'sentry' database"
-	$(CREATEDB) -h 127.0.0.1 -U postgres -E utf-8 sentry || true
-
-apply-migrations:
-	@./scripts/do.sh apply-migrations
-
-reset-db: drop-db create-db apply-migrations
+bootstrap \
```
Comment (armenzg) [inline Makefile:4]: Josh, this looks very nice. Thank you!

File: scripts/lib.sh [hunk 7]
```diff
@@ -85,3 +88,68 @@ develop() {
...
+
+init-config() {
+    sentry init --dev
+}
+
+run-dependent-services() {
+    sentry devservices up
+}
+
+create-db() {
```
Comment (armenzg) [inline scripts/lib.sh:100]: Porting `create-db` and `drop-db` are the only part of this PR that required few tries to port it.

File: Makefile (hunk 6 above)
Comment (joshuarli) [reply Makefile:4]: :)

File: Makefile (hunk 5 above)
Comment (joshuarli) [reply Makefile:19]: upgrade-pip should be in the venv, but you should just remove it. It's just used by install-py-dev.

Comment (joshuarli) [reply Makefile:19]: Oh, well, it's used by getsentry actually, so just move it to normal do.sh.

File: Makefile [hunk 8]
```diff
@@ -199,6 +151,7 @@ lint-js:
 	@echo ""
 
 
+# https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html
 .PHONY: bootstrap \
```
Comment (joshuarli) [inline Makefile:155]: Just remove all these phony targets except `build`. They're not realistic, and the effort to maintain this just isn't worth it. See https://github.com/getsentry/sentry/pull/24612#pullrequestreview-618189325.

**Review by joshuarli — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (armenzg) [reply Makefile:19]: Unfortunately, it's also used as part of the CI outside of a venv, thus, need to set `SENTRY_NO_VENV_CHECK`.

File: Makefile (hunk 8 above)
Comment (armenzg) [reply Makefile:155]: Why `build` is worth keeping around? Is there something special about it? Or do we need at least a list of 1?

Comment (billyvg) [reply Makefile]: Guessing here, but out of all of these targets, `build` has the highest chance of being an actual file in the fs.

File: Makefile (hunk 5 above)
Comment (joshuarli) [reply Makefile:19]: Oh.

File: Makefile (hunk 8 above)
Comment (joshuarli) [reply Makefile]: Billy's right.

File: Makefile (hunk 5 above)
Comment (joshuarli) [reply Makefile:19]: COuld be something worth improving, seems kinda fragile.

**Review by joshuarli — APPROVED** [review]

**Pull request merged**